Platformsh.php

```diff
@@ -457,4 +458,48 @@ protected function generatePassword($password)
             ]
         );
     }
+
+    /**
+     * Execute compilation command.
+     * Doing multi-tenant compilation always, because there is known issue with single-tenant one
+     */
+    protected function doCompilation()
+    {
+        $this->log("Removing old compilation files");
+        $this->execute('rm -rf var/di/*');
+        $this->log("Doing code compilation");
+        $this->execute(
+            "cd bin/; /usr/bin/php ./magento setup:di:compile-multi-tenant"
+        );
+    }
```

Comment:
Any compilation belongs in the build process. Ideally, all these files should be read-only. Writing PHP files in production is far from desirable.